docs/pages/how-it-works/overview.mdx

```diff
@@ -0,0 +1,35 @@
+---
+title: How it works
+sidebar_label: Overview
+---
+
+Devpod provides the ability to provision workspaces on any infrastructure. It does so by wrapping your conventential CLI tools such as kubectl, docker, gcloud etc to deploy your development environment 
+and set up everything required to run the dev container. While creating the workspace devpod deploys an agent to the machine running the container as well as to the container itself to provide useful 
+functions such as port forwarding, log streaming and more. Doing so it provides a control plane across your development environment. DevPod can be thought of as a minimal service mesh, specifically
+designed to connect your local environment securely to your development container running in your cloud infrastructure.
+
+Devpod uses a client-agent architecture, where the client deploys it's own agent to host various servers, such as a grpc server or SSH server. 
+In this regard the system is not unlike a browser server architecture where the front end is deployed and executed on a remote host. There are several improvements this brings to our specific context:
+ - There can be no conflict of versions between client and server, since you install only one version of the client
+ - There is no infrastructure to manage for users
+ 
+To simplify debugging, devpod connects your local shell with the agent's STDIO so you can see what's happening locally and in the container at all times.
+ 
+All traffic between your local machine
+and the workspace is tunneled through, ensuring secure communication using the cloud vendor's API.
+
+Below is a high level overview of how DevPod uses your local environment, a source repo and a devcontainer to deploy your workspace to the cloud.
+
+<figure>
+  <img src="/docs/media/c4_workspaces.png" alt="DevPod Architecture" />
+  <figcaption>DevPod - Component Diagram</figcaption>
+</figure>
+
+When you installed devpod a default "context" was created for you. This context is like a kubernetes context where devpod will use the config values you specify based on the context you are currently using.
```

Comment:
let's remove the context if possible. I think it's mostly confusing to take it into consideration here. 
If we want to explain it a bit more in depth, let's create a dedicated page